.github/actions/reporter/src/coverage.ts

```diff
@@ -0,0 +1,52 @@
+import { readFile } from 'fs/promises';
+
+import * as core from '@actions/core';
+
+import { GitHubCommentBuilder } from './comment';
+
+interface Stat {
+  total: number;
+  covered: number;
+  skipped: number;
+  pct: number;
+}
+
+interface Stats {
+  lines: Stat;
+  statements: Stat;
+  functions: Stat;
+  branches: Stat;
+  branchesTrue?: Stat;
+}
+
+interface Coverage {
+  total: Stats;
+  [key: string]: Stats;
+}
+
+async function parseFile(coveragePath: string) {
+  const file = await readFile(coveragePath, 'utf-8');
+  return JSON.parse(file) as Coverage;
+}
```

Comment:
Выглядит как штука, которую можно вынести в `./shared.js` переопределив типы вот так

```suggestion
async function parseFile<T>(path: string) {
  const file = await readFile(path, 'utf-8');
  return JSON.parse(file) as T;
}
```

Reply:
✅ вынес